### PR TITLE
Add integration test for CodeActSWEAgent

### PR DESCRIPTION
**Pull request by yufansong**
As title. Follow the test of `CodeActAgent`, add test `test_edits`, `test_ipython`, `test_opython_module`, `test_write_simple_script`. Because we disable the `browsing` feature in `CodeActSWEAgent`, I didn't add `test_browse_internet` test.

### CODE REVIEW

**Comment by tobitege** [comment]
Please do not merge before https://github.com/OpenDevin/OpenDevin/pull/2346 , please!

**Comment by yufansong** [comment]
> Please do not merge before #2346 , please!

Why? 
1. It seems you change the prompts in `CodeActAgent`, which will not influence my test on `CodeActSWEAgent`
2. You change something on `agentskill`, but I think these two pr not related? If any conflict happended, we can regenerate the test case.

**Comment by tobitege** [comment]
Oh, I just got a little panic but the SWE version doesn't use agentskills, right?
If so, ignore my previous post.

**Comment by yufansong** [comment]
> Oh, I just got a little panic but the SWE version doesn't use agentskills, right? If so, ignore my previous post.

I checked the logic, SWE version used the agentskills. If my PR merge before your PR, you need to regenerate the test log. If your PR merge before my PR, I need to regenerate the test log. I think regenerate is just one command and easy to execute. I think it is not a big deal for both of us. If I was wrong, feel free to correct me.

I see your PR still blocked by many other things. So personally spearking, I don't want to block by your PR. Because #2368 also waiting for this one.

**Comment by tobitege** [comment]
My PR isn't blocked and been waiting for hours to get merged so I can continue with other things.

**Comment by yufansong** [comment]
> My PR isn't blocked and been waiting for hours to get merged so I can continue with other things.

I see this in your title: (wait for 2085!). And I know this work may still take some time.
If you think you PR can be merged now, I can help you click merge.

**Comment by tobitege** [comment]
> > My PR isn't blocked and been waiting for hours to get merged so I can continue with other things.
> 
> I see this in your title: (wait for 2085!). And I know this work may still take some time. If you think you PR can be merged now, I can help you click merge.

Right, just now fixed the title. Would really appreciate if you wouldn't mind with a review once the tests are done.
Thank you :)

**Comment by yufansong** [comment]
> > > My PR isn't blocked and been waiting for hours to get merged so I can continue with other things.
> > 
> > 
> > I see this in your title: (wait for 2085!). And I know this work may still take some time. If you think you PR can be merged now, I can help you click merge.
> 
> Right, just now fixed the title. Would really appreciate if you wouldn't mind with a review once the tests are done. Thank you :)

Get it. I see you ask for other maintainers review. I can wait for it. I guess they will response in 24h. I can wait. :)

**Comment by tobitege** [comment]
this is weird in the tests log:
```
open_file("bad.txt")
16:09:11 - OBSERVATION
**IPythonRunCellObservation**
bash: line 1: execute_cli: command not found
```

File: tests/integration/test_agent.py
Comment (enyst) [inline tests/integration/test_agent.py:105]:
```suggestion
    reason='currently only CodeActAgent and CodeActSWEAgent have IPython (Jupyter) execution by default',
```

File: tests/integration/test_agent.py
Comment (enyst) [inline tests/integration/test_agent.py:148]:
```suggestion
    reason='currently only CodeActAgent and CodeActSWEAgent have IPython (Jupyter) execution by default',
```

File: tests/integration/mock/CodeActSWEAgent/test_ipython_module/prompt_008.log
Comment (xingyaoww) [inline tests/integration/mock/CodeActSWEAgent/test_ipython_module/prompt_008.log:550]: Is this normal?

File: tests/integration/mock/CodeActSWEAgent/test_ipython_module/prompt_008.log
Comment (xingyaoww) [inline tests/integration/mock/CodeActSWEAgent/test_ipython_module/prompt_008.log:574]: Huh, the environment is different from bash terminal and IPython shell.. This is tricky :(

**Review by xingyaoww — APPROVED** [review]
Mostly LGTM. There's some issues with `test_ipython_module` we need to think about how to improve - basically when `pip install` is run on bash, the jupyter kernel won't get automatically rebooted. Ideally we can throw an error message suggesting "If you think the package is installed, please reboot your jupyter kernel, etc".

But in the long run, we will probably have two separate Python environments:
1. IPython kernel which is mainly used for `agentskills` tools
2. Bash terminal, where the agent can actually switch between different virtual environments
We probably need to think of a way for agent to know this better

File: tests/integration/mock/CodeActSWEAgent/test_ipython_module/prompt_008.log
Comment (yufansong) [reply tests/integration/mock/CodeActSWEAgent/test_ipython_module/prompt_008.log:550]: Humm, `Command -1` looks strange.

File: tests/integration/mock/CodeActSWEAgent/test_ipython_module/prompt_008.log
Comment (yufansong) [reply tests/integration/mock/CodeActSWEAgent/test_ipython_module/prompt_008.log:574]: Record the issue here #2402

**Pull request merged**
